pkg/cmd/roachtest/test_runner.go

```diff
@@ -1111,6 +1119,10 @@ func (r *testRunner) runTest(
 		c.grafanaTags = []string{vm.SanitizeLabel(runID), vm.SanitizeLabel(testRunID), vm.SanitizeLabel(c.Name())}
 	}
 
+	if err := maybeSetupCpuProfiling(ctx, t, c); err != nil {
```

Comment:
It seems to me that this is trying to run SQL on the cluster before the test has started? I don't think that would work but lmk if I'm missing something and it works for you.

Instead I think it makes more sense to pass them as cluster settings when `c.Start` is called. If you look at `c.StartE`, it should be straightforward to add something like:

```
	if roachtestflags.ForceCpuProfile {
		settings.ClusterSettings["server.cpu_profile.duration"] = "20s"
		// and so on...
	}
```

This should also do what you want wrt only applying the settings to the system tenant.

Reply:
In the TC run I'm seeing: 

```
failed to set up cpu profiling for cluster [teamcity-18020145-1733261817-02-n4cpu16] - dial tcp 34.138.117.140:26257: connect: connection refused
=== RUN   tpccbench/nodes=3/cpu=16  [metrics: https://go.crdb.dev/roachtest-grafana/teamcity-18020145/tpccbench-nodes-3-cpu-16/1733261946957/1733263746957
failed to set up cpu profiling for cluster [teamcity-18020145-1733261817-04-n4cpu4] - dial tcp 34.73.183.24:26257: connect: connection refused
```

Reply:
It looks like it failed to run in one of the tests that I executed in that build, but not for the others.